File: src/game/entity/models/Direction.java
package game.entity.models;

import game.hardwareinput.Command;

public enum Direction {
    LEFT,
    RIGHT,
    UP,
    DOWN;

    public Command toCommand() {
        switch (this) {
            case UP:
                return Command.MOVE_UP;
            case DOWN:
                return Command.MOVE_DOWN;
            case LEFT:
                return Command.MOVE_LEFT;
            case RIGHT:
                return Command.MOVE_RIGHT;
        }
        return Command.MOVE_UP;
    }

    public Direction opposite() {
        switch (this) {
            case UP:
                return DOWN;
            case DOWN:
                return UP;
            case LEFT:
                return RIGHT;
            case RIGHT:
                return LEFT;
            default:
                return UP;
        }

    }


}

File: src/game/domain/world/domain/entity/geo/Direction.kt
package game.domain.world.domain.entity.geo

import game.input.game.Command

enum class Direction {
    DOWN, LEFT, UP, RIGHT;

    fun toCommand(): Command {
        return when (this) {
            UP -> Command.MOVE_UP
            DOWN -> Command.MOVE_DOWN
            LEFT -> Command.MOVE_LEFT
            RIGHT -> Command.MOVE_RIGHT
        }
    }

    fun opposite(): Direction {
        return when (this) {
            UP -> DOWN
            DOWN -> UP
            LEFT -> RIGHT
            RIGHT -> LEFT
        }
    }
}